qucs/qucs/qucs.cpp

```diff
@@ -2284,6 +2284,10 @@ void QucsApp::slotSelectSubcircuit(const QModelIndex &idx)
 
   QString filename = idx.sibling(idx.row(), 0).data().toString();
   QString note = idx.sibling(idx.row(), 1).data().toString();
+  int idx_pag = DocumentTab->currentIndex();
+  QString tab_titl = "";
+  if (idx_pag>=0) tab_titl = DocumentTab->tabText(idx_pag);
+  if (filename == tab_titl ) return; // Forbid to paste subcircuit into itself.
```

Comment:
I thought about this possibility, but since you need to have a Project open to insert a subcircuit symbol by clicking on its filename in the Project View tree it won't be possible to have files with the same names but different absolute paths. All the files for a project reside in the same subdirectory.
Sure, the solution is a bit of a hack...